src/components/Button/index.js

```diff
@@ -1,59 +1,166 @@
-import React from "react";
-import PropTypes from "prop-types";
-import styled, { css } from "styled-components"
-import { colors } from "../../styles";
-import { darken } from "../../utils/color";
-
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components'
+import { colors } from '../../styles';
+import { darken, lighten } from '../../utils/color';
 
 const types = {
   default: {
     background: colors.colorPrimary,
-    borderColor: darken(colors.colorPrimary, 10),
+    color: colors.colorWhite,
+    display: 'inline-block',
   },
   primary: {
     background: colors.colorSecondary,
-    borderColor: darken(colors.colorSecondary, 10),
+    color: colors.colorWhite,
   },
   success: {
     background: colors.colorCoreGreen,
-    borderColor: darken(colors.colorCoreGreen, 10),
-  }
+    color: colors.colorWhite,
+  },
+  danger: {
+    background: colors.colorCoreRed,
+    color: colors.colorWhite,
+  },
+  warning: {
+    background: colors.colorCoreYellow,
+    color: colors.colorWhite,
+  },
+  simple: {
+    background: colors.colorWhite,
+    borderColor: colors.colorCoreLightGray,
+    color: colors.colorCoreLightGray,
+  },
+  link: {
+    background: 'transparent',
+    color: colors.colorCoreGray,
+  },
+  block: {
+    width: '100%',
+    display: 'block',
+  },
 };
 
+const sizes = {
+  large: {
+    padding: '0.688em 2.063em',
+    fontSize: '0.875em',
+  },
+  medium: {
+    padding: '0.665em 2em',
+    fontSize: '0.688em',
+  },
+  small: {
+    padding: '0.5em 1.375em',
+    fontSize: '0.625em',
+  },
+};
+
+const ButtonLink = styled.a`${props => css`
```

Comment:
too many duplicated code in ButtonLink and ButtonStyled. use extend from styled-component